### PR TITLE
[Enhancement] OutputProperty of PartitionTopN returns children property

### PR DESCRIPTION
**Pull request by ZiheLiu**
Signed-off-by: ZiheLiu <ziheliu1024@gmail.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Relates to #17553.

## Problem Summary(Required) ：

When hitting rank window partition optimization, there is a `PartitionTopN` before `Sort->Window`.
However, `PartitionTopN` doesn't inherent output property from children, which results in unuseless ExchangeSinkNode.

For exmaple, `OlapScanNode->Sort->Window` will be changed to `OlapScanNode->PartitionTopN->Exchange  --> Sort->Window`

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OutputPropertyDeriver.java [hunk 1]
```diff
@@ -342,8 +342,19 @@ public PhysicalPropertySet visitPhysicalTopN(PhysicalTopNOperator topN, Expressi
             } else {
                 outputProperty = new PhysicalPropertySet(new SortProperty(topN.getOrderSpec()));
             }
-        } else {
+        } else if (topN.getPartitionByColumns() == null) {
             outputProperty = PhysicalPropertySet.EMPTY;
+        } else {
+            List<Integer> partitionColumnRefSet = new ArrayList<>();
+            topN.getPartitionByColumns().forEach(c ->
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OutputPropertyDeriver.java:349]: how about a `flatMap` ?

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OutputPropertyDeriver.java (hunk 1 above)
Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OutputPropertyDeriver.java]: Good suggestion.

**Comment by ZiheLiu** [comment]
run starrocks_admit_test

**Comment by ZiheLiu** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 7 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/OutputPropertyDeriver.java|6|7|85.71%|[352]|

**Review by satanson — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**